internal/tiltfile/k8s.go

```diff
@@ -887,3 +888,57 @@ func newK8sObjectID(e k8s.K8sEntity) k8sObjectID {
 func (s *tiltfileState) k8sContext(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
 	return starlark.String(s.kubeContext), nil
 }
+
+func (s *tiltfileState) allowK8SContexts(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var contexts starlark.Value
+	if err := starlark.UnpackArgs(fn.Name(), args, kwargs,
+		"contexts", &contexts,
+	); err != nil {
+		return nil, err
+	}
+
+	for _, c := range starlarkValueOrSequenceToSlice(contexts) {
+		switch val := c.(type) {
+		case starlark.String:
+			s.whitelistedK8SContexts = append(s.whitelistedK8SContexts, k8s.KubeContext(val))
+		default:
+			return nil, fmt.Errorf("allow_k8s_contexts contexts must be a string or a sequence of strings; found a %T", val)
+
+		}
+	}
+
+	s.whitelistedK8SContexts = append(s.whitelistedK8SContexts)
+
+	return starlark.None, nil
+}
+
+func (s *tiltfileState) validateK8SContext(kubeConfig *api.Config) error {
+	contextName := kubeConfig.CurrentContext
+	clusterName := kubeConfig.Contexts[contextName].Cluster
+	server := kubeConfig.Clusters[clusterName].Server
+	url, err := url.Parse(server)
+	if err != nil {
+		return errors.Wrapf(err, "running in kube context %q, which specifies kube api url %q. error parsing that url", contextName, server)
+	}
+
+	ipaddr, err := net.ResolveIPAddr("ip", url.Hostname())
+	if err != nil {
+		return errors.Wrapf(err, "running in kube context %q, which specifies kube api url %q. error resolving host %q", contextName, url, url.Host)
+	}
+
+	if ipaddr.IP.IsLoopback() {
+		return nil
+	}
+
+	for _, c := range s.whitelistedK8SContexts {
+		if c == k8s.KubeContext(contextName) {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("Current kube context %q is neither whitelisted nor known to be local. If this is not the "+
```

Comment:
This error message is pretty inscrutable

I like looking at Chrome error messages for inspiration
https://lifehacker.com/report-suspicious-sites-to-google-using-this-chrome-ext-1835650376

how about something like

"
Stop! X is a production cluster.
If you're sure that this is a dev cluster, add add
allow_k8s_contexts('X')
to your Tiltfile
```

No need to even give them an API link

Reply:
That's better, thanks.

Here's what it looks like now:
![image](https://user-images.githubusercontent.com/7453991/62636967-898b5e80-b908-11e9-83eb-2ae9be1954df.png)